### ISSUE TITLE
VAMC Phone number/contact accessibility

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#5710]
## Description

Some of the telephone numbers are breaking onto multiple lines. I think this could be from the CMS adding tags where there shouldn't have been. This could create a situation where keyboard users cannot launch their preferred visual phone application correctly. This will also have adverse effect on screen reader users for the same reason.

I found 5-6 on the https://www.va.gov/pittsburgh-health-care/locations/pittsburgh-va-medical-center-university-drive/ page, so it will probably be a wider effort to discover all of these instances. Screenshot attached below.

## Type of Issue
Content

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Acceptance Criteria

* [ ] The telephone links are formatted correctly and will receive focus without breaking across multiple lines or tab stops

## Environment

* MacOS Mojave
* Chrome latest
* https://www.va.gov/pittsburgh-health-care/locations/pittsburgh-va-medical-center-university-drive/ (and possibly others)

## Steps to Recreate

1. Enter https://www.va.gov/pittsburgh-health-care/locations/pittsburgh-va-medical-center-university-drive/ in browser
2. Press `TAB` until the `Radiology` accordion has keyboard focus (yellow halo)
3. Press `ENTER` to open the accordion
4. Press `TAB` several more times until you see a telephone number that receives the yellow focus halo twice 

 **adding this note for more info**
- The phone number is 412-360-6105. The first halo is around the area code, the second around the rest of the number. Phone numbers which are behaving as expected have a single halo around the entire number, including the area code. 
- In this instance, it is both phone numbers within the Radiology accordion (under contact info and appointments)

## Possible Fixes (optional)
Removing the extra spaces in the CMS and/or the tag markup will make these links work correctly:

```html
<p class="va-address-block">
  <strong>Phone:&nbsp;</strong>
  <a href="tel:%20+14128223000">412-360-6105</a>
  <strong>Hours:</strong>&nbsp;Monday – Friday, 8:00 a.m. to 4:30 p.m. ET
</p>
```

## WCAG or Vendor Guidance (optional)

* [Link Purpose (In Context): Understanding SC 2.4.4](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-refs.html)

## Screenshots or Trace Logs

![Screen Shot 2020-02-10 at 4 43 01 PM](https://user-images.githubusercontent.com/934879/74198421-eb65b180-4c26-11ea-85b3-a0fa273a9da8.png)

**Comment by mmiddaugh** [comment]
In checking into this, I found that the phone number in the Library accordion did not get focus. 
![image](https://user-images.githubusercontent.com/55411834/74692778-9f25ee80-519d-11ea-94bd-2d1d2af51a22.png)

The numbers in the accordians within the Phone Directory on https://www.va.gov/pittsburgh-health-care/contact-us/ also do not receive focus. 

I assume all phone numbers should. Is this correct, @1Copenut?

**Comment by mmiddaugh** [comment]
https://www.va.gov/pittsburgh-health-care/locations/h-john-heinz-iii-department-of-veterans-affairs-medical-center/
The focus is shown twice on the second phone number for the Veterans Recovery Center. Second halo is between the 1 and 2 of "1287"

![image](https://user-images.githubusercontent.com/55411834/74693041-e2cd2800-519e-11ea-98d1-e6ef6a6b3c45.png)

**Comment by mmiddaugh** [comment]
Issues observed on https://www.va.gov/pittsburgh-health-care/locations/beaver-county-va-clinic/

1. Both instances of 412-360-1210 under LGBT Veteran Care receive second halo after area code
2. Both instances of 724-709-6005 under Primary Care have second halo around a space immediately after the phone number

**Comment by mmiddaugh** [comment]
Issues observed on https://www.va.gov/pittsburgh-health-care/locations/belmont-county-va-clinic/

Both instances of 740-695-9321 are receiving a second halo within or after the area code. This phone number is repeated in several sections: 
- HIV/hepatitis
- Intimate partner abuse 
- Military sexual trauma
* note: this number behaves as expected in other areas

Both instances of 412-360-1210 under LGBT Veteran Care receive second halo after the area code

**Comment by mmiddaugh** [comment]
Per conversation with @kevwalsh, this could be handled by adding an address block.

**Comment by kevwalsh** [comment]
The address block component (and/or phone number component), which will make it easier for editors to add phone number links, is probably months away. I can imagine that being a vsa-facilities initiative, as it's primarily facilities products that will leverage it, or a "core" CMS component, since it will be shared among many products. 

**In the meantime:** 
These look like content problems within VA Pittsburgh health care. Similar ownership problems: I'm not sure if that's a VAMC upgrade team responsibility, a VSA-facilities responsibility, or the responsibility of the Public Affairs Officers who own this content in Pittsburgh. But that one is more clear, CMS team is not managing the content itself any product.

**Comment by ethanteague** [comment]
@mmiddaugh @davidconlon 
Confirmed that these broken phone links are being caused during content entry, not by cms.

**Scenario 1:** 
Radiology accordion. Looks like the entry person entered an incorrect number when clicking the link button over the phone number the first time, and in correcting the number, left the old bad number on the area code. They then copy-pasted this into the wysiwyg a second time. Place of entry: `/node/629/edit`
![image](https://user-images.githubusercontent.com/2404547/79049877-2ae35680-7bf4-11ea-8e43-5e0956c8ab06.png)

**Scenario 2:** 
Library accordion. Content entry person entered a phone number, but didn't enter a link via the wysiwyg button. Place of entry: `node/84/edit` in `Location services "Prepare for your visit"`
![image](https://user-images.githubusercontent.com/2404547/79049884-359deb80-7bf4-11ea-98de-941dfcbb6e1f.png)

**Scenario 3:** 
Contact info. Content entry person entered phone numbers, but didn't enter links via the wysiwyg button. Place of entry: `node/255/edit` in `Main content Wysiwyg`
![image](https://user-images.githubusercontent.com/2404547/79049888-3df62680-7bf4-11ea-86d8-41277c89cf5b.png)

**Scenario 4:** 
Records. Content entry person entered phone numbers, but didn't enter links via the wysiwyg button. Place of entry: `node/221/edit` in `Main content Wysiwyg`
![image](https://user-images.githubusercontent.com/2404547/79049889-43537100-7bf4-11ea-9417-1cc6b693c038.png)


**Possible ways to prevent this from occurring in the future:**
 - Add a cms phone field below the wysiwyg that will take care of the link handling.
Drawback to this approach is that this number will be disconnected from the existing field, and putting it back together with the wysiwyg content would be a mess. This approach would require content entry team training to prevent phone numbers from being entered in the wysiwyg.
 - Tighten up the content entry review process - prior to draft moving to published status, reviewer to confirm that phone links in wysiwyg work.

**Possible ways to address the incorrectly formatted numbers currently in the cms:**
### Manually: 
 - e.g., content entry team revisits items they have created, and reformats the phone numbers and resaves the content.

### Programmatically:
- The Drupal way = create an update hook, process the data during a code release. This approach would be complicated, and any time we bulk save content programmatically, it creates revision confusion.
- The js/metalsmith way = create a filter that processes the handful of wysiwyg fields where we are seeing broken numbers, and format the numbers to be properly wrapped in link tags. This approach is less complicated, but will probably take a day or two to implement.

**Comment by mmiddaugh** [comment]
@jenstrickland can this issue be closed?

**Comment by jenstrickland** [comment]
If there is time…

* **Consider** putting more space between links. The numbers under Audiology are well-spaced, but those under Beneficiary Travel are too close. This also happens in other spots down the page. Best practices recommend 24px between interactive elements. 

* Consider adding the [new phone number pattern](https://dsva.slack.com/archives/C0NGDDXME/p1589812701321700), of adding the attribute pattern: `aria-label="8 0 0. 5 5 5. 5 5 5 5." for each phone number. This is a usability enhancement for screen readers. Instead of a string of hundreds and thousands quickly read, this pattern reads one number at a time with a small pause. 

**Example 1**
 ![Screen Shot 2020-05-18 at 11 21 35 AM](https://user-images.githubusercontent.com/57469/82231250-e6d22880-98fa-11ea-8d99-0bb09f95e784.png)

**Example 2**
![Screen Shot 2020-05-18 at 11 21 50 AM](https://user-images.githubusercontent.com/57469/82231282-ef2a6380-98fa-11ea-8b6a-dad27136d1ea.png)

**Example 3**
![Screen Shot 2020-05-18 at 11 21 59 AM](https://user-images.githubusercontent.com/57469/82231291-f5204480-98fa-11ea-97df-f6bd889ce52a.png)
